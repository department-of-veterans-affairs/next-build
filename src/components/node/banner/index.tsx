import { useRef, useEffect, useState } from 'react'
import { VaBanner } from '@department-of-veterans-affairs/component-library/dist/react-bindings'
import { isEmpty } from 'lodash'
import { recordEvent } from '@/utils/recordEvent'
import { NodeMetaInfo, NodeResourceType } from '@/types/node'

const BannerComponent = ({ node }): JSX.Element => {
  const [isClicked, setIsClicked] = useState(false)
  const analyticsRef = useRef(null)

  useEffect(() => {
    function handler(event) {
      if (analyticsRef.current?.contains(event.target)) {
        setIsClicked(true)
      }
    }
    window.addEventListener('click', handler)
    return () => window.removeEventListener('click', handler)
  }, [])

  if (isClicked) {
    recordEvent({
      eventCategory: 'Banner',
      eventAction: 'Click',
      eventLabel: node.title,
    })
  }

  if (isEmpty(node)) return

  return (
    <VaBanner
      id={node.id}
      role="va-banner"
      showClose={node.field_dismissible_option != 'perm'}
      headline={node.title}
      type={node.field_alert_type}
      visible={true}
      windowSession={node.field_dismissible_option == 'dismiss-session'}
    >
      <div
        ref={analyticsRef}
        dangerouslySetInnerHTML={{ __html: node?.body?.processed }}
      />
    </VaBanner>
  )
}
export default BannerComponent

export const Meta: NodeMetaInfo = {
  resource: NodeResourceType.Banner,
  component: BannerComponent,
  params: null,
  collection: true,
}
