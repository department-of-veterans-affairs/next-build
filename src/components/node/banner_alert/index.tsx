import { isEmpty } from 'lodash'
import { useRef, useEffect, useState } from 'react'
import { recordEvent } from '@/utils/recordEvent'
import { regionBaseURL, trackLinks } from '@/utils/helpers'
import { VaBanner } from '@department-of-veterans-affairs/component-library/dist/react-bindings'

const BannerAlert = ({ node }): JSX.Element => {
  const [isClicked, setIsClicked] = useState(false)
  const analyticsRef = useRef(null)

  useEffect(() => {
    function handler(event) {
      if (analyticsRef.current?.contains(event.target)) {
        setIsClicked(true)
      }
    }
    window.addEventListener('click', handler)
    return () => window.removeEventListener('click', handler)
  }, [])

  if (isEmpty(node)) return

  const hideOnSubpages = node.field_alert_inheritance_subpages
  const alertType =
    node.field_alert_type === 'information' ? 'info' : node.field_alert_type
  const region = '/' + regionBaseURL(node.path.alias)
  const lastArg = node.path.alias.substring(node.path.alias.lastIndexOf('/'))
  const bodyEventData = {
    event: 'nav-alert-box-link-click',
    'alert-box-status': alertType,
    'alert-box-headline': node.title,
    'alert-box-headline-level': '3',
    'alert-box-background-only': 'false',
    'alert-box-closeable': 'false',
  }
  let body = trackLinks(node.field_body.processed, bodyEventData)
  let outputStatus = false
  let statusUrl = ''

  if (isClicked) {
    recordEvent({
      eventCategory: 'Banner',
      eventAction: 'View',
      eventLabel: node.title,
    })
  }

  node.field_banner_alert_vamcs?.map((vamcs) => {
    if (region == vamcs.field_office.path.alias) {
      outputStatus = true
    }
    if (hideOnSubpages && lastArg != region && lastArg != '/operating-status') {
      outputStatus = false
    }
    statusUrl = vamcs.path.alias
  })

  if (node.field_alert_operating_status_cta && statusUrl.length) {
    const ctaEventData = {
      event: 'nav-warning-alert-box-content-link-click',
      alertBoxHeading: `${node.title}`,
    }

    body += `<p>
          <a href='${statusUrl}' onClick='recordEvent(${JSON.stringify(
      ctaEventData
    )})'>
            Get updates on affected services and facilities
          </a>
      </p>`
  }

  if (node.field_alert_find_facilities_cta) {
    body += `
      <p>
        <a href="/find-locations">Find other VA facilities near you</a>
      </p>`
  }

  return (
    <VaBanner
      id={node.id}
      role="va-banner"
      showClose={node.field_alert_dismissable}
      headline={node.title}
      type={node.field_alert_type}
      visible={outputStatus}
      windowSession={node.field_alert_dismissable == 'dismiss-session'}
    >
      <div ref={analyticsRef} dangerouslySetInnerHTML={{ __html: body }} />
    </VaBanner>
  )
}
export default BannerAlert