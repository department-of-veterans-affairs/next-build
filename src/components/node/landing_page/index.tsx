import Link from 'next/link'

// Output a landing_page node as a link to itself, plus its teaser text.
const BenefitsHubLink = ({ node }) => {
  if (node) {
    return (
      <>
        <p className="vads-u-margin--0">
          <strong>
            <Link href={node.path?.alias ? node.path?.alias : ' '} passHref>
              <a>{node.field_home_page_hub_label}</a>
            </Link>
          </strong>
        </p>
        <p className="vads-u-margin--0">{node.field_teaser_text}</p>
      </>
    )
  }
}

// This should not be the default; we will have a full display eventually.
// Only here to meet required default rules.
export default BenefitsHubLink
