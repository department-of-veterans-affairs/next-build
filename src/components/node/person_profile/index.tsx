import { v4 as uuidv4 } from 'uuid'
import Image from '@/components/image'
import { DEV_PATH } from '@/lib/constants'
import { StaffProfiles } from '@/components/paragraph/staff_profile'

export const PersonProfile = ({ node }): JSX.Element => {
  if (!node) return
  const {
    title,
    field_body: fieldBody,
    field_email_address: fieldEmailAddress,
    field_phone_number: fieldPhoneNumber,
    field_suffix: suffix,
    field_description: fieldDescription,
    field_complete_biography_create: fieldCompleteBiographyCreate,
    field_intro_text: fieldIntroText,
    field_media: fieldMedia,
    field_photo_allow_hires_download: fieldPhotoAllowHiresDownload,
    field_complete_biography: fieldCompleteBiography,
  } = node

  const thumbnail = DEV_PATH + fieldMedia?.thumbnail?.uri?.url
  const image = DEV_PATH + fieldMedia?.image?.uri?.url
  const meta = fieldMedia?.thumbnail?.resourceIdObjMeta

  return (
    <div key={uuidv4()}>
      <h3>Person Profile</h3>
      <article className="usa-content">
        <div className="usa-grid usa-grid-full vads-u-margin-bottom--2 vads-u-display--flex vads-u-flex-direction--column medium-screen:vads-u-flex-direction--row">
          {fieldMedia?.thumbnail && (
            <div className="vads-u-margin-bottom--2 medium-screen:vads-u-margin-bottom--0 vads-u-margin-right--3">
              <Image
                className="person-profile-detail-page-image"
                src={thumbnail}
                alt={meta?.alt}
                title={meta.title}
                width={meta?.width}
                height={meta?.height}
              />
            </div>
          )}
          <div className="vads-u-display--flex vads-u-flex-direction--column">
            <h1 className="vads-u-font-size--xl vads-u-margin-bottom--0p5">
              {title}
              {suffix && [suffix]}
            </h1>
            {fieldDescription && (
              <p className="vads-u-font-size--lg vads-u-margin-bottom--0p5">
                {fieldDescription}
              </p>
            )}
            {fieldEmailAddress && (
              <p className="vads-u-font-size--lg vads-u-margin-bottom--0p5">
                <span className="vads-u-font-weight--bold">Email: </span>
                <a target="blank" href="mailto:{{ fieldEmailAddress }}">
                  {fieldEmailAddress}
                </a>
              </p>
            )}
            {fieldPhoneNumber && (
              <p
                className="
                vads-u-font-weight--regular
                vads-u-margin--0
                vads-u-margin-bottom--0p5"
              >
                <span className="vads-u-font-weight--bold">Phone: </span>
                <a href="tel:{{ fieldPhoneNumber }}">{fieldPhoneNumber}</a>
              </p>
            )}
          </div>
        </div>
        {fieldCompleteBiographyCreate && (
          <div className="va-introtext vads-u-margin-bottom--2">
            <p className="va-introtext vads-u-margin-bottom--0">
              {fieldIntroText}
            </p>
            <div
              className="vads-u-margin-bottom--2"
              dangerouslySetInnerHTML={{ __html: fieldBody.processed }}
            />
          </div>
        )}
        {fieldPhotoAllowHiresDownload && (
          <div
            className="vads-u-align-content--flex-end va-c-margin-top--auto vads-u-margin-bottom--2"
            id="download-full-size-photo-link"
          >
            <i
              className="va-c-social-icon fas fa-download"
              aria-hidden="true"
            ></i>
            <a href={image} download>
              Download full size photo
            </a>
          </div>
        )}
        {fieldCompleteBiography && (
          <div
            className="vads-u-align-content--flex-end va-c-margin-top--auto vads-u-margin-bottom--2"
            id="download-full-size-photo-link"
          >
            <i
              className="va-c-social-icon fas fa-download"
              aria-hidden="true"
            ></i>
            <div className="vads-u-align-content--flex-end va-c-margin-top--auto vads-u-margin-bottom--2">
              <i className="va-c-social-icon fas fa-download"></i>
              <a href={fieldCompleteBiography.url} download>
                Download full bio (PDF)
              </a>
            </div>
          </div>
        )}
      </article>
      <hr />
    </div>
  )
}

export const StaffNewsProfile = ({ node }): JSX.Element => {
  if (!node) return

  const { id, title, field_description: fieldDescription } = node

  return (
    <div className="vads-u-font-size--sm vads-u-margin-bottom--2p5">
      <div className="authored-by-line vads-u-margin-bottom--0p5 vads-u-font-weight--bold">
        By {title}
        {fieldDescription ? `, ${fieldDescription}` : null}
      </div>
    </div>
  )
}

export const StaffProfile = ({ paragraph }) => {
  return (
    <div key={uuidv4()}>
      <h3>Staff Profile</h3>
      <StaffProfiles paragraph={paragraph} />
    </div>
  )
}
