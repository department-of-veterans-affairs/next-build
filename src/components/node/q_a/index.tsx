/**
 * ### Overview
 * Resource QA is a node that displays a question and answer section.
 *
 * Resources QA expects nodes of type {@link NodeQA}.
 *
 * ### View modes
 * Full page: {@link NodeQAFull}
 *
 * ### Examples
 * @see https://va.gov/resources/how-are-pension-benefits-and-disability-compensation-different
 *
 */

/** These types/packages will import into all node components. */
import { NodeQA, NodeResourceType, NodeMetaInfo } from '@/types/node'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

/** These component includes are specific to this component. */
import Container from '@/components/container'
import { BenefitsHubLinks } from '@/components/partials/benefitHubsLinks'
import { Paragraph } from '@/lib/delegators/Paragraph'

/**
 * These components expect NodeQA as their input.
 */
type NodeQaProps = {
  node: NodeQA
}

export const ResourceWrapper = ({
  title,
  fieldAnswer,
  buttons,
  tags,
  linkTeaser,
  benefits,
}) => {
  return (
    <div id="content" className="interior" data-template="node-q_a">
      <Container className="container">
        <main className="va-l-detail-page">
          <div className="usa-grid usa-grid-full">
            <div className="usa-width-three-fourths">
              <div className="usa-content">
                {/** Add Search Widget Here */}
                <article className="vads-u-padding-x--1 large-screen:vads-u-padding-x--0">
                  <h1>{title}</h1>
                  <div dangerouslySetInnerHTML={fieldAnswer} />
                  <ul className="vads-u-margin-top--3 vads-u-margin-bottom--3 usa-unstyled-list">
                    {buttons}
                  </ul>
                  {tags}
                  {/*Section title*/}
                  <h2 className="vads-u-margin-y--0 vads-u-font-size--h3">
                    Related information
                  </h2>
                  {/*Links*/}
                  <ul className="usa-unstyled-list">{linkTeaser}</ul>
                  {benefits}
                </article>
              </div>
            </div>
          </div>
        </main>
      </Container>
    </div>
  )
}

export const Resources = ({ node }: NodeQaProps) => {
  /** Type narrowing; if we've managed to end up here with the wrong data, return. */
  if (node?.type !== NodeResourceType.QuestionAnswer) return

  const resources = {
    benefits: <BenefitsHubLinks nodes={node?.field_related_benefit_hubs} />,
    tags: <Paragraph paragraph={node?.field_tags} />,
    teaser: node?.field_related_information.map((paragraphLinkTeaser) => (
      <Paragraph
        key={paragraphLinkTeaser?.id}
        paragraph={paragraphLinkTeaser}
        componentParams={[{ boldTitle: false }, { sectionHeader: '' }]}
      />
    )),
    buttons: node.field_buttons.map((paragraphButton) => (
      <Paragraph key={paragraphButton.id} paragraph={paragraphButton} />
    )),
  }

  return (
    <ResourceWrapper
      title={node?.title}
      tags={resources?.tags}
      benefits={resources?.benefits}
      buttons={resources?.buttons}
      fieldAnswer={{
        __html: node?.field_answer?.field_wysiwyg?.processed,
      }}
      linkTeaser={resources?.teaser}
    ></ResourceWrapper>
  )
}

/** All nodes end with NodeMetaInfo: the name of the resource, the name of the component, and the parameters necessary for calling the resource. */
const params = new DrupalJsonApiParams().addInclude([
  'field_buttons',
  'field_related_information',
  'field_answer',
  'field_tags, field_tags.field_audience_beneficiares',
  'field_related_benefit_hubs',
  'field_contact_information',
])

/** Export information necessary to identify the component and query it.
 * See {@link NodeMetaInfo}
 */
export const Meta: NodeMetaInfo = {
  resource: NodeResourceType.QuestionAnswer,
  component: Resources,
  params: params,
}
