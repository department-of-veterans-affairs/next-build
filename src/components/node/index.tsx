import { Meta as NewsStoryMeta } from '@/components/node/news_story'
import { NodeTypes } from '@/types/node'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

/** General NodeProps to pass nodes into node components. */
interface NodeProps {
  node: NodeTypes
  viewMode?: string
}

/** We expect each Node component to define and export information about itself to help with discoverability. */
export interface NodeMetaInfo {
  resource: string
  component: ({ node, viewMode, ...props }: NodeProps) => JSX.Element
  params: DrupalJsonApiParams
}

/** Collect all imported node meta information. */
const nodeMetaIn: NodeMetaInfo[] = [NewsStoryMeta]

/** This interface enforces that the Node meta information is indexable by type. */
interface NodeMetaOut {
  [resource: string]: {
    component: ({ node }: NodeProps) => JSX.Element
    params: DrupalJsonApiParams
  }
}

/** Converts the meta information into a form indexed by resource type. Very possibly overwrought. */
export const nodeMeta: NodeMetaOut = nodeMetaIn.reduce((acc, current) => {
  const { resource, component, params } = current
  acc[resource] = {
    component: component,
    params: params,
  }
  return acc
}, {})

/** Generalized component. Look up the component the data requires and use it to render. */
export function Node({ node, viewMode = 'full', ...props }: NodeProps) {
  if (!node) {
    return null
  }

  const Component = nodeMeta[node.type].component

  if (!Component) {
    return null
  }

  return <Component node={node} viewMode={viewMode} {...props} />
}
