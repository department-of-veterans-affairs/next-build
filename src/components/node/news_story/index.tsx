/** These types/packages will import into all node components. */
import { NodeMetaInfo } from '@/components/node'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

/** These component includes are specific to this component. */
import { MediaImage } from '@/components/media'
import { StaffNewsProfile } from '@/components/node/person_profile'
import { NodeNewsStory } from '@/types/node'

type NodeNewsStoryProps = {
  node: NodeNewsStory
  viewMode?: string
}

/** Full page news story. */
export const NewsStoryFull = ({ node }: NodeNewsStoryProps) => {
  /** Type narrowing; if we've managed to end up here with the wrong data, return. */
  if (node.type !== 'node--news_story') return
  const dateTime = new Date(node.created).toLocaleDateString('en-us', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  })
  return (
    <>
      <div id="content" className="interior">
        <main className="va-l-detail-page va-facility-page">
          <div className="usa-grid usa-grid-full">
            {/* nav here */}
            <div className="usa-width-three-fourths">
              <article className="usa-content">
                <h1>{node.title}</h1>
                <MediaImage media={node.field_media} imageStyle={'2_1_large'} />
                <div className="vads-u-font-size--sm vads-u-margin-bottom--2p5">
                  {node.field_image_caption}
                </div>
                <StaffNewsProfile node={node.field_author} />
                <div className="created-line vads-u-margin-bottom--2p5">
                  <time dateTime="{ dateTime }">{dateTime}</time>
                </div>

                {/* {% include "src/site/facilities/story_social_share.drupal.liquid" %} */}

                <div className="usa-grid usa-grid-full vads-u-margin-bottom--2">
                  <div className="va-introtext">
                    <p>{node.field_intro_text}</p>
                  </div>
                  <div
                    className="full-story"
                    dangerouslySetInnerHTML={{
                      __html: node.field_full_story.processed,
                    }}
                  ></div>
                </div>
                {/* {% if fieldListing.entity.entityUrl.path %}
                        <a onClick="recordEvent({ event: 'nav-secondary-button-click' });" className="vads-u-display--block vads-u-margin-bottom--7" href="{{ fieldListing.entity.entityUrl.path }}" id="news-stories-listing-link">See all stories</a>
                    {% endif %} */}
              </article>
            </div>
          </div>
        </main>
      </div>
    </>
  )
}

/** General News Story component. Allows choice of different display components by the caller. */
export const NewsStory = ({ node, viewMode, ...props }: NodeNewsStoryProps) => {
  switch (viewMode) {
    case 'full':
      return <NewsStoryFull node={node} {...props} />
      break

    default:
      return null
  }
}

/** All nodes end with NodeMetaInfo: the name of the resource, the name of the component, and the parameters necessary for calling the resource. */
const params = new DrupalJsonApiParams().addInclude([
  'field_media',
  'field_media.image',
  'field_author',
])
export const Meta: NodeMetaInfo = {
  resource: 'node--news_story',
  component: NewsStory,
  params: params,
}
