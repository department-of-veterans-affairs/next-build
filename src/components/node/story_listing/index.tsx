/**
 * ### Overview
 * Story Listing represents an individual story within a Facility. These are used for human-interest articles.
 *
 * Story Listing expects nodes of type {@link NodeStoryListing}.
 *
 * ### View modes
 * Teaser: {@link StoryListing}
 *
 * ### Examples
 * @see https://va.gov/pittsburgh-health-care/stories/
 *


/** These types/packages will import into all node components. */
import { NodeMetaInfo, NodeResourceType } from '@/types/node'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

/**
 * These components expect NodeStoryListing as their input.
 */

import Container from '@/components/container'
import { NewsStory } from '@/components/node/news_story'
import { isValidData } from '@/utils/helpers'

/** General Story Listing component. Allows choice of different display components by the caller. */
const StoryListing = ({ node, additionalNode }): JSX.Element => {
  if (!isValidData(node || additionalNode)) {
    return
  }

  return (
    <>
      {node.map((storyListing) => (
        <div key={storyListing.id} className="usa-grid usa-grid-full">
          {/*<div className="usa-width-three-fourths">*/}
          <h1>{storyListing.title}</h1>
          <div className="vads-l-grid-container--full">
            <div className="va-introtext">
              {storyListing.field_intro_text && (
                <p className="events-show">{storyListing.field_intro_text}</p>
              )}
              {additionalNode.length === 0 ||
                (additionalNode.length === null && (
                  <div className="clearfix-text">No stories at this time.</div>
                ))}
              <Container className="container">
                <ul className="usa-unstyled-list">
                  {additionalNode.map((newsStoryTeaser) => (
                    <NewsStory
                      node={newsStoryTeaser}
                      key={newsStoryTeaser.id}
                      viewMode="teaser"
                    />
                  ))}
                </ul>
              </Container>
            </div>
          </div>
          {/*</div>*/}
        </div>
      ))}
    </>
  )
}

export default StoryListing

/** All nodes end with NodeMetaInfo: the name of the resource, the name of the component, and the parameters necessary for calling the resource. */
const params = new DrupalJsonApiParams()
  .addFilter('status', '1')
  .addSort('created', 'DESC')

const newsStory = new DrupalJsonApiParams()
  .addInclude(['field_media', 'field_media.image', 'field_listing'])
  .addPageLimit(10)

/** Export information necessary to identify the component and query it.
 * See {@link NodeMetaInfo}
 */

export const Meta: NodeMetaInfo = {
  resource: NodeResourceType.StoryListing,
  component: StoryListing,
  params: params,
  additionalNode: NodeResourceType.NewsStory,
  additionalParams: newsStory,
  collection: true,
}
