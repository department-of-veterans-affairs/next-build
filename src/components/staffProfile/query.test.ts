/**
 * @jest-environment node
 */

import { NodePersonProfile } from '@/types/drupal/node'
import { queries } from '@/lib/drupal/queries'
import mockData from './mock.personProfile.json'
import { RESOURCE_TYPES } from '@/lib/constants/resourceTypes'

//eslint-disable-next-line
const personProfileMock: NodePersonProfile | any = mockData

describe('Person profile returns formatted data', () => {
  test('outputs formatted data', () => {
    const formattedData = personProfileMock
    expect(
      queries.formatData(RESOURCE_TYPES.STAFF_PROFILE, {
        entity: formattedData,
      })
    ).toMatchSnapshot()
  })
})
