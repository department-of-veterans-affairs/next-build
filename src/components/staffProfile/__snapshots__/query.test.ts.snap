// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Person profile returns formatted data outputs formatted data 1`] = `
{
  "administration": {
    "entityId": null,
    "name": null,
  },
  "body": null,
  "breadcrumbs": undefined,
  "completeBiography": null,
  "completeBiographyCreate": 1,
  "description": "Director, Audiology and Speech Pathology Program Associate Director for Research, GRECC",
  "emailAddress": null,
  "entityId": 812,
  "entityPath": "/pittsburgh-health-care/staff-profiles/patrick-j-doyle",
  "firstName": "Patrick J.",
  "id": "fe42bc63-0933-40bb-978a-ef951fa75684",
  "introText": null,
  "lastName": "Doyle",
  "lastUpdated": "2019-08-27T17:49:03+00:00",
  "media": {
    "alt": "Patrick J. Doyle",
    "height": 129,
    "id": "5bcbdaee-f93f-4b2f-8af5-963e4a547d93",
    "links": {
      "1_1_square_medium_thumbnail": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/1_1_square_medium_thumbnail/public/2019-08/pat_doyle_0.jpg?h=46bc4a38",
        "meta": {
          "linkParams": {
            "height": "240",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "240",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
      "2_1_large": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/2_1_large/public/2019-08/pat_doyle_0.jpg",
        "meta": {
          "linkParams": {
            "height": "512",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "1024",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
      "2_1_medium_thumbnail": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/2_1_medium_thumbnail/public/2019-08/pat_doyle_0.jpg",
        "meta": {
          "linkParams": {
            "height": "240",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "480",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
      "2_3_medium_thumbnail": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/2_3_medium_thumbnail/public/2019-08/pat_doyle_0.jpg",
        "meta": {
          "linkParams": {
            "height": "480",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "320",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
      "3_2_medium_thumbnail": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/3_2_medium_thumbnail/public/2019-08/pat_doyle_0.jpg",
        "meta": {
          "linkParams": {
            "height": "320",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "480",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
      "7_2_medium_thumbnail": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/7_2_medium_thumbnail/public/2019-08/pat_doyle_0.jpg",
        "meta": {
          "linkParams": {
            "height": "300",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "1050",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
      "full_content_width": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/full_content_width/public/2019-08/pat_doyle_0.jpg",
        "meta": {
          "linkParams": {
            "height": "129",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "100",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
      "large": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/large/public/2019-08/pat_doyle_0.jpg",
        "meta": {
          "linkParams": {
            "height": "129",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "100",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
      "self": {
        "href": "https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/file/file/5bcbdaee-f93f-4b2f-8af5-963e4a547d93",
      },
      "viewport_width": {
        "href": "http://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/sites/default/files/styles/viewport_width/public/2019-08/pat_doyle_0.jpg",
        "meta": {
          "linkParams": {
            "height": "129",
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
            ],
            "width": "100",
          },
          "rel": [
            "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
          ],
        },
      },
    },
    "title": "",
    "type": "media--image",
    "width": 100,
  },
  "menu": {
    "depth": 5,
    "link": {
      "label": "Patrick J. Doyle",
      "links": [],
      "url": {
        "path": "/",
      },
    },
    "parent": {
      "label": "Leadership",
      "links": [],
      "url": {
        "path": "/pittsburgh-health-care/about-us/leadership",
      },
    },
  },
  "metatags": null,
  "moderationState": "published",
  "office": {
    "changed": "2022-08-31T15:49:34+00:00",
    "created": "2019-05-06T03:30:18+00:00",
    "default_langcode": true,
    "drupal_internal__nid": 318,
    "drupal_internal__vid": 679006,
    "field_administration": {
      "id": "87832236-1e54-4ce3-8141-8dec27c8a9a7",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 12,
      },
      "type": "taxonomy_term--administration",
    },
    "field_appointments_online": true,
    "field_clinical_health_services": [],
    "field_description": "Find a health facility near you at VA Pittsburgh health care, and manage your health online. Our health care teams are deeply experienced and guided by the needs of Veterans, their families, and caregivers.",
    "field_facebook": {
      "options": [],
      "title": "VA Pittsburgh health care Facebook",
      "uri": "https://www.facebook.com/VAPHS",
    },
    "field_flickr": {
      "options": [],
      "title": "VA Pittsburgh health care Flickr",
      "uri": "https://www.flickr.com/photos/vapittsburgh/sets",
    },
    "field_govdelivery_id_emerg": "USVHA_396",
    "field_govdelivery_id_news": "USVHA_94",
    "field_instagram": null,
    "field_intro_text": "At VA Pittsburgh Healthcare System, our health care teams are deeply experienced and guided by the needs of Veterans, their families, and caregivers. Find a health facility near you, and manage your health online. Sign up for community events and updates.",
    "field_media": {
      "id": "de919547-d935-46da-a220-9f0646f2c0e9",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 57,
      },
      "type": "media--image",
    },
    "field_meta_tags": null,
    "field_operating_status": {
      "options": [],
      "title": "VA Pittsburgh health care operating status",
      "uri": "entity:node/1010",
    },
    "field_other_va_locations": [
      "vc_0211V",
      "vc_0233V",
      "vc_0220V",
      "nca_925",
      "vba_311",
    ],
    "field_related_links": {
      "id": "c47aaf53-cd32-47b7-84af-9c004926b211",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 3943,
        "target_revision_id": 694430,
      },
      "type": "paragraph--list_of_link_teasers",
    },
    "field_system_menu": {
      "id": "83b58096-0fa0-4e12-895b-6dc58a2b9c88",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": "pittsburgh-health-care",
      },
      "type": "menu--menu",
    },
    "field_twitter": {
      "options": [],
      "title": "VA Pittsburgh health care Twitter",
      "uri": "https://twitter.com/VAPittsburgh",
    },
    "field_va_health_connect_phone": null,
    "field_vamc_ehr_system": "vista",
    "field_vamc_system_official_name": "VA Pittsburgh Healthcare System",
    "id": "2bddb1a7-6fb1-4503-838d-9c2fcb51c46a",
    "langcode": "en",
    "links": {
      "self": {
        "href": "https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/node/health_care_region_page/2bddb1a7-6fb1-4503-838d-9c2fcb51c46a?resourceVersion=id%3A679006",
      },
    },
    "metatag": null,
    "moderation_state": "published",
    "node_type": {
      "id": "d611ade3-cc27-4359-8af1-6fba3384200f",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": "health_care_region_page",
      },
      "type": "node_type--node_type",
    },
    "path": {
      "alias": "/pittsburgh-health-care",
      "langcode": "en",
      "pid": 503,
    },
    "promote": false,
    "relationshipNames": [
      "node_type",
      "revision_uid",
      "uid",
      "field_administration",
      "field_clinical_health_services",
      "field_media",
      "field_related_links",
      "field_system_menu",
    ],
    "resourceIdObjMeta": {
      "drupal_internal__target_id": 318,
    },
    "revision_log": "Set "System Menu" field accordingly (see va.gov-cms/#9520).",
    "revision_timestamp": "2022-07-18T13:30:31+00:00",
    "revision_translation_affected": true,
    "revision_uid": {
      "id": "183f10f5-b0dc-4a10-9955-cfa65b6f15af",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 1454,
      },
      "type": "user--user",
    },
    "status": true,
    "sticky": false,
    "title": "VA Pittsburgh health care",
    "type": "node--health_care_region_page",
    "uid": {
      "id": "fd71292f-3dc7-409d-b0e8-500204cb005a",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 65,
      },
      "type": "user--user",
    },
  },
  "phoneNumber": {
    "entityId": undefined,
    "extension": null,
    "id": undefined,
    "label": "Appointments",
    "number": "800-274-8387",
    "phoneType": "tel",
    "type": undefined,
  },
  "photoAllowHiresDownload": false,
  "published": true,
  "suffix": "Ph.D.",
  "title": "Patrick J. Doyle",
  "type": "node--person_profile",
  "vamcTitle": "VA Pittsburgh health care",
}
`;
